src/pubsub.c

```diff
@@ -603,20 +612,24 @@ void punsubscribeCommand(client *c) {
 
 /* This function wraps pubsubPublishMessage and also propagates the message to cluster.
  * Used by the commands PUBLISH/SPUBLISH and their respective module APIs.*/
-int pubsubPublishMessageAndPropagateToCluster(robj *channel, robj *message, int sharded) {
-    int receivers = pubsubPublishMessage(channel, message, sharded);
-    if (server.cluster_enabled) clusterPropagatePublish(channel, message, sharded);
+int pubsubPublishMultiMessagesAndPropagateToCluster(robj *channel, robj **messages, int count, int sharded) {
+    int receivers = pubsubPublishMultiMessages(channel, messages, count, sharded);
+    if (server.cluster_enabled) clusterPropagatePublish(channel, messages, count, sharded);
     return receivers;
 }
 
+int pubsubPublishMessageAndPropagateToCluster(robj *channel, robj *message, int sharded) {
+    return pubsubPublishMultiMessagesAndPropagateToCluster(channel, &message, 1, sharded);
+}
+
 /* PUBLISH <channel> <message> */
 void publishCommand(client *c) {
     if (server.sentinel_mode) {
         sentinelPublishCommand(c);
         return;
     }
 
-    int receivers = pubsubPublishMessageAndPropagateToCluster(c->argv[1], c->argv[2], 0);
+    int receivers = pubsubPublishMultiMessagesAndPropagateToCluster(c->argv[1], &c->argv[2], 1, 0);
```

Comment:
This change is not needed, since you have already routed `pubsubPublishMessageAndPropagateToCluster` to `pubsubPublishMultiMessagesAndPropagateToCluster`